app/blueprints/assessments/routes.py

```diff
@@ -411,36 +362,493 @@ def get_sorted_application_overviews(application_overviews, column, reverse=Fals
-        countries=all_application_locations.countries,
-        regions=all_application_locations.regions,
-        local_authorities=all_application_locations._local_authorities,
-        migration_banner_enabled=Config.MIGRATION_BANNER_ENABLED,
-        dpi_filters=dpi_filters,
-        users=["All", "Not assigned"] + fund_user_aliases,
+        form=form,
+    )
+
+
+@assessment_bp.route(
+    "/assign/<fund_short_name>/<round_short_name>/assessors",
+    methods=["POST"],
+)
+@check_access_fund_short_name_round_sn
+def assessor_type_list(fund_short_name: str, round_short_name: str):
+    selected_assessments = request.form.getlist("selected_assessments")
+    assessor_role = request.form.getlist("assessor_role")[0]
+    selected_users = request.form.getlist("selected_users")
+
+    form = AssessorChoiceForm()
+
+    if form.validate_on_submit():
+        return redirect(
+            url_for(
+                "assessment_bp.assignment_overview",
+                fund_short_name=fund_short_name,
+                round_short_name=round_short_name,
+            ),
+            307,
+        )
+
+    fund = get_fund(
+        fund_short_name,
+        use_short_name=True,
+        ttl_hash=get_ttl_hash(Config.LRU_CACHE_TIME),
+    )
+    if not fund:
+        return redirect("/assess/assessor_tool_dashboard/")
+
+    round = get_round(
+        fund_short_name,
+        round_short_name,
+        use_short_name=True,
+        ttl_hash=get_ttl_hash(Config.LRU_CACHE_TIME),
+    )
+
+    round_details = {
+        "assessment_deadline": round.assessment_deadline,
+        "round_title": round.title,
+        "fund_name": fund.name,
+        "fund_short_name": fund_short_name,
+        "round_short_name": round_short_name,
+        "is_expression_of_interest": round.is_expression_of_interest,
+    }
+
+    thread_executor = ContextAwareExecutor(
+        max_workers=4,
+        thread_name_prefix="fund-dashboard-request",
+        flask_app=current_app,
+    )
+
+    # The first call is to get the location data such as country, region and local_authority
+    # from all the existing applications (i.e without search parameters as we don't want to filter
+    # the stats at all).  see https://dluhcdigital.atlassian.net/browse/FS-3249
+    future_all_applications_metadata = thread_executor.submit(
+        get_application_overviews, fund.id, round.id, ""
     )
 
+    # Get all the users for the fund
+    future_users_for_fund = thread_executor.submit(
+        get_users_for_fund,
+        fund.short_name,
+        None,  # round_short_name,
+        True,
+        False,
+    )
+
+    all_applications_metadata = future_all_applications_metadata.result()
+    users_for_fund = future_users_for_fund.result()
+
+    if assessor_role.lower() == "lead_assessor":
+        selectable_users = [
+            user
+            for user in users_for_fund
+            if any("LEAD_ASSESSOR" in role for role in user["roles"])
```

Comment:
shall we add constant values for these ?

Reply:
Sure will do that.